.github/workflows/publish.yml

```diff
@@ -0,0 +1,36 @@
+name: publish
+
+on:
+  push:
+    branches: [master]
+    tags: ['*']
+
+jobs:
+  publish-scm-1:
+    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: tarantool/rocks.tarantool.org/github-action@master
+        with:
+          auth: ${{ secrets.ROCKS_AUTH }}
+          files: smtp-scm-1.rockspec
+
+  publish-tag:
+    if: startsWith(github.ref, 'refs/tags/')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      # Make a release
+      - run: printf '%s=%s\n' TAG "${GITHUB_REF##*/}" >> "${GITHUB_ENV}"
+      - run: sed -E
+          -e "s/branch = '.+'/tag = '${{ env.TAG }}'/g"
+          -e "s/version = '.+'/version = '${{ env.TAG }}-1'/g"
+          smtp-scm-1.rockspec >> smtp-${{ env.TAG }}-1.rockspec
```

Comment:
Verified YAML syntax:

```
tarantool> print(require('yaml').decode(require('http.client').get('https://raw.githubusercontent.com/tarantool/smtp/a14240a73954897f7d48c2816345ba94bb9385ac/.github/workflows/publish.yml').body).jobs['publish-tag'].steps[3].run)

sed -E -e "s/branch = '.+'/tag = '${{ env.TAG }}'/g" -e "s/version = '.+'/version = '${{ env.TAG }}-1'/g" smtp-scm-1.rockspec >> smtp-${{ env.TAG }}-1.rockspec
```

Verified the sed expressions:

```diff
$ diff -u smtp-scm-1.rockspec <(sed -E -e "s/branch = '.+'/tag = '0.0.6'/g" -e "s/version = '.+'/version = '0.0.6-1'/g" smtp-scm-1.rockspec)
--- smtp-scm-1.rockspec	2021-04-19 21:24:57.075189726 +0300
+++ /dev/fd/63	2021-04-28 18:12:10.169930214 +0300
@@ -4,12 +4,12 @@
 -- version of the package; it's mandatory, but we don't use it in Tarantool;
 -- instead, provide below a specific branch in the package's repository at
 -- GitHub and set version to some stub value, e.g. 'scm-1'
-version = 'scm-1'
+version = '0.0.6-1'
 
 -- url and branch of the package's repository at GitHub
 source  = {
     url    = 'git://github.com/tarantool/smtp.git';
-    branch = 'master';
+    tag = '0.0.6';
 }
 
 -- general information about the package;
```

Everything works as expected, nice!